### PR TITLE
TOOLS/PERF: Fix memory leak of msg size list

### PR DESCRIPTION
**Pull request by dmitrygx**
## What

Fix memory leak of msg size list + minor code cleanups

## Why ?

it fixes the memory leak, e.g. it's detected by ASan
```
==1439==ERROR: LeakSanitizer: detected memory leaks

Direct leak of 8 byte(s) in 1 object(s) allocated from:
    #0 0x7fda383c212e in __interceptor_calloc ../../../../source/gcc-9.2.0/libsanitizer/asan/asan_malloc_linux.cc:153
    #1 0x407671 in init_test_params /labhome/dmitrygla/work_auto/ucx/src/tools/perf/perftest.c:608
    #2 0x40939c in parse_opts /labhome/dmitrygla/work_auto/ucx/src/tools/perf/perftest.c:873
    #3 0x40ca38 in main /labhome/dmitrygla/work_auto/ucx/src/tools/perf/perftest.c:1680
    #4 0x7fda35d923d4 in __libc_start_main (/lib64/libc.so.6+0x223d4)

SUMMARY: AddressSanitizer: 8 byte(s) leaked in 1 allocation(s).
```

## How ?

Release the memory pointed by `ctx->params.super.msg_size_list` before doing `safe_recv()` of the params from the client

### CODE REVIEW

**Comment by dmitrygx** [comment]
@brminich @hoopoepg could you review pls?

File: src/tools/perf/perftest.c [hunk 1]
```diff
@@ -1075,13 +1075,16 @@ static ucs_status_t setup_sock_rte(struct perftest_context *ctx)
 
         close(sockfd);
 
+        free(ctx->params.super.msg_size_list);
```
Comment (yosefe) [inline src/tools/perf/perftest.c:1078]: where was this allocated before? can you pls add comment to explain why need to free here?

File: src/tools/perf/perftest.c [hunk 2]
```diff
@@ -1585,31 +1588,33 @@ static ucs_status_t run_test_recurs(struct perftest_context *ctx,
...
+        free(ctx->test_names[depth]);
+        ctx->test_names[depth] = NULL;
     }
 
     if (status == UCS_ERR_NO_ELEM) {
         status = UCS_OK;
     }
 
     free(params.super.msg_size_list);
+    params.super.msg_size_list = NULL;
```
Comment (yosefe) [inline src/tools/perf/perftest.c:1617]: maybe we don't need the free() in line 1616 now? and also this? since it's don'e in lines 1605-1606?

Comment (dmitrygx) [reply src/tools/perf/perftest.c:1617]: yes, we could avoid this `free()`, but need to reorganize code a bit
done

File: src/tools/perf/perftest.c (hunk 1 above)
Comment (dmitrygx) [reply src/tools/perf/perftest.c:1078]: it was allocated in `init_test_params()` for server and client sides
added the comment

**Comment by dmitrygx** [comment]
@yosefe ok to merge?

**Pull request merged**